### DISCUSSION TITLE
How do I enable deletion protection on resources with Terragrunt and infrastructure-live/Reference Architecture

### DISCUSSION THREAD

**Discussion by iantybo** [gruntwork-io/knowledge-base#629] [How do I] [answered]
I understand I can use things like [CODEOWNERS](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) at the repo level, and [prevent_destroy](https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#prevent_destroy) at the Terraform resource level, but is there a Terragrunt mechanism that I can use at the module level? For example, a way to make sure `./*/*/networking/vpc*` and `./*/*/networking/vpc-mgmt*` can't be deleted, even when those changes have been checked into the main branch of the repo?  

I am considering setting up Gruntwork Pipelines following this [Knowledge Base answer](https://github.com/gruntwork-io/knowledge-base/discussions/349) as well if there is a path there, but I'd like to know if there is a native Terragrunt way to do this. 

---

<ins datetime="2023-01-10T23:12:06Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109785">Tracked in ticket #109785</a></p>
</ins>

**Comment by Etiene** [comment, marked as answer]
Yes, there is an attribute for that: https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#prevent_destroy
